aws_lambda_builders/workflows/nodejs_npm/workflow.py

```diff
@@ -97,48 +68,46 @@ def actions_without_bundler(self, source_dir, artifacts_dir, scratch_dir, manife
         ]
 
         if self.download_dependencies:
-            # installed the dependencies into artifact folder
-            install_action = NodejsNpmWorkflow.get_install_action(
-                source_dir, artifacts_dir, subprocess_npm, osutils, self.options
+            # install the dependencies into artifact folder
+            self.actions.append(
+                NodejsNpmWorkflow.get_install_action(source_dir, artifacts_dir, subprocess_npm, osutils, self.options)
+            )
+
+        artifacts_cleanup_actions = [
+            NodejsNpmrcCleanUpAction(artifacts_dir, osutils=osutils),
+            NodejsNpmLockFileCleanUpAction(artifacts_dir, osutils=osutils),
+        ]
+
+        # if no dependencies dir, just cleanup artifacts and we're done
+        if not self.dependencies_dir:
+            self.actions += artifacts_cleanup_actions
+            return
+
+        # if we downloaded dependencies, update dependencies_dir
+        if self.download_dependencies:
+            # clean up the dependencies folder first
+            self.actions.append(CleanUpAction(self.dependencies_dir))
+            # if combine_dependencies is set, we should keep dependencies and source code in the artifact folder
+            # while copying the dependencies. Otherwise we should separate the dependencies and source code
+            dependencies_dir_update_action = (
+                CopyDependenciesAction if self.combine_dependencies else MoveDependenciesAction
             )
-            actions.append(install_action)
-
-            # if dependencies folder exists, copy or move dependencies from artifact folder to dependencies folder
-            # depends on the combine_dependencies flag
-            if self.dependencies_dir:
-                # clean up the dependencies folder first
-                actions.append(CleanUpAction(self.dependencies_dir))
-                # if combine_dependencies is set, we should keep dependencies and source code in the artifact folder
-                # while copying the dependencies. Otherwise we should separate the dependencies and source code
-                if self.combine_dependencies:
-                    actions.append(CopyDependenciesAction(source_dir, artifacts_dir, self.dependencies_dir))
-                else:
-                    actions.append(MoveDependenciesAction(source_dir, artifacts_dir, self.dependencies_dir))
-        else:
-            # if dependencies folder exists and not download dependencies, simply copy the dependencies from the
-            # dependencies folder to artifact folder
-            if self.dependencies_dir and self.combine_dependencies:
-                actions.append(CopySourceAction(self.dependencies_dir, artifacts_dir))
-            else:
-                LOG.info(
-                    "download_dependencies is False and dependencies_dir is None. Copying the source files into the "
-                    "artifacts directory. "
-                )
```

Comment:
I thought this message was a bit weird and not totally accurate, so I removed it. The message could be printed if `dependencies_dir` is **not** None, but `combine_dependencies` is False. Let me know if you think I should bring this message back.

Reply:
Note that the conditions that lead to this message being output is not a valid SAM CLI use case anyway (shouldn't be very common): 
https://github.com/aws/aws-lambda-builders/pull/272/files#r701280071